### PR TITLE
Out 256 Task Details Page UI

### PR DESCRIPTION
**Pull request by aatbip**
Ref: [OUT-256](https://linear.app/copilotplatforms/issue/OUT-256/task-details-ui-design)

- Adds complete task details page 
- Adds setup to redux toolkit
- Adds reusable components such as DatePicker, status and assignee selectors, and attachment

### CODE REVIEW

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/app/ui/CreateTask.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
+'use client'
+
+import { PrimaryBtn } from '@/components/buttons/PrimaryBtn'
+import { SecondaryBtn } from '@/components/buttons/SecondaryBtn'
+import { SelectorType } from '@/components/inputs/Selector'
+import StatusSelector from '@/components/inputs/Selector'
+import AssigneeSelector from '@/components/inputs/Selector'
```
[the remark this answers is not in the document]
Comment (saugatmaharjan) [reply src/app/ui/CreateTask.tsx:7]: nice catch @rrojan 👍

File: package.json [hunk 2]
```diff
@@ -36,10 +36,14 @@
...
     "@mui/icons-material": "^5.15.15",
     "@mui/material": "^5.15.6",
+    "@mui/x-date-pickers": "^7.1.1",
     "@prisma/client": "^5.12.0",
     "@reduxjs/toolkit": "^2.2.3",
+    "@types/date-fns": "^2.6.0",
     "@vercel/postgres": "^0.8.0",
     "copilot-node-sdk": "^2.0.0",
+    "date-fns": "^3.6.0",
+    "dayjs": "^1.11.10",
```
[the remark this answers is not in the document]
Comment (aatbip) [reply package.json:46]: Removed the unnecessary dependency `date-fns`

File: src/app/ui/CreateTask.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aatbip) [reply src/app/ui/CreateTask.tsx:7]: addressed in the commits below

File: src/app/ui/CreateTask.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+              value={assigneeValue}
+              selectorType={SelectorType.ASSIGNEE_SELECTOR}
+              buttonContent={
+                <Typography variant="bodySm" lineHeight="16px" sx={{ color: (theme) => theme.color.gray[600] }}>
+                  {(assigneeValue as IAssignee)?.name || 'No Assignee'}
+                </Typography>
+              }
+            />
+          </Stack>
+        </Stack>
```
[the remark this answers is not in the document]
Comment (aatbip) [reply src/app/ui/CreateTask.tsx:107]: `Selector` component is already a reusable component being used in `CreateTask.tsx` and `src/app/detail/ui/Sidebar.tsx`. Giving another level of abstraction is unnecessary and has the props to be drilled.

File: src/app/ui/TaskBoard.tsx [hunk 4]
```diff
@@ -2,15 +2,16 @@
 
 import { useState, useCallback } from 'react'
 import update from 'immutability-helper'
-import { Modal, Stack } from '@mui/material'
+import { Box, Modal, Stack } from '@mui/material'
 import { TaskCard } from '@/components/cards/TaskCard'
 import { TaskColumn } from '@/components/cards/TaskColumn'
 import { Droppable } from '@/hoc/Droppable'
-import { CreateTask } from '../components/CreateTask'
+import { CreateTask } from '../ui/CreateTask'
```
[the remark this answers is not in the document]
Comment (aatbip) [reply src/app/ui/TaskBoard.tsx:9]: addressed in the next commit

File: src/components/inputs/Selector.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+              if (selectorType === SelectorType.ASSIGNEE_SELECTOR) {
+                return (option as IAssignee).name as string
+              } else return option as string
+            }}
+            groupBy={(option: unknown) => {
+              if (selectorType === SelectorType.ASSIGNEE_SELECTOR) {
+                return (option as IAssignee).type
+              } else {
+                return ''
+              }
```
[the remark this answers is not in the document]
Comment (aatbip) [reply src/components/inputs/Selector.tsx:74]: addressed in the next commit

File: src/app/ui/NewTaskForm.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      <Stack
+        direction="row"
+        alignItems="center"
+        justifyContent="end"
+        sx={{
+          border: (theme) => `1px solid ${theme.color.borders.borderDisabled}`,
+        }}
+      >
+        <AppMargin size={SizeofAppMargin.MEDIUM} py="12px">
+          <Close
```
Comment (rrojan) [inline src/app/ui/NewTaskForm.tsx:39]: Please reference this comment https://github.com/copilot-platforms/tasks-app/pull/5#pullrequestreview-2000955150

File: src/components/inputs/Selector.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
+import { Avatar, Box, Stack, Typography } from '@mui/material'
+import { SecondaryBtn } from '../buttons/SecondaryBtn'
+import { StyledAutocomplete } from './Autocomplete'
```
Comment (rrojan) [inline src/components/inputs/Selector.tsx:3]: Let's avoid using relative imports :bow:

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/components/inputs/Selector.tsx (hunk 7 above)
Comment (aatbip) [reply src/components/inputs/Selector.tsx]: addressed in ce20cc7d9007779f481a24b6ae555a8ac039d0bc commit

File: src/app/ui/NewTaskForm.tsx (hunk 6 above)
Comment (rrojan) [reply src/app/ui/NewTaskForm.tsx:39]: Sounds great!

**Pull request merged**